### PR TITLE
[V7] Bump min deployment target to iOS 16

### PR DESCRIPTION
**Pull request by scannillo**
### Summary of changes

- Bump min deployment target of SDK to iOS 16+
- Update package manager manifest files, framework targets, and sample apps

### Checklist

- [X] Added a changelog entry

### Authors
@scannillo

### CODE REVIEW

**Review by saperi22 — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 ## unreleased
 * BraintreeCore
   * Fix analytics bug where sessionID value in analytics payload was inaccurate; send separate FPTI POST requests per unique sessionID
+* Breaking Changes
```
Comment (jaxdesmarais) [inline CHANGELOG.md:6]: Not sure if it'd be easier to create a v7 section of the changelog for now. May make it easier to reason about what is a v7 vs v6 change with merge conflicts too.

**Review by jaxdesmarais — APPROVED** [review]
Do we also want to bump the Xcode minimum to 16+ and Swift 6 with this?

**Comment by scannillo** [comment]
> Do we also want to bump the Xcode minimum to 16+ and Swift 6 with this?

Looks like we have separate tix for these

File: CHANGELOG.md (hunk 1 above)
Comment (scannillo) [reply CHANGELOG.md:8]: Good call, updated!

**Pull request merged**